pkg/generic/thrift/write.go

```diff
@@ -397,11 +453,38 @@ func writeList(ctx context.Context, val interface{}, out thrift.TProtocol, t *de
 	if length == 0 {
 		return out.WriteListEnd()
 	}
-	writer, err := nextWriter(l[0], t.Elem, opt)
-	if err != nil {
-		return err
+	var (
+		writer writer
+		init   interface{}
+		err    error
+	)
+	existNil := false
+	for _, elem := range l {
+		if writer != nil && existNil {
```

Comment:
Just try to get `writer` in this loop, no need to judge  whether existNil, nil value can be replaced by zeroValue if we meet it in the next writing loop.

Reply:
Just to confirm, if both nil and non-nil values exist, we get the writer with nextWriter(), right?

Reply:
Right.